### DISCUSSION TITLE
Can't compile programs for attiny1616 any more, it was working before

### DISCUSSION THREAD

**Discussion by bdelapen** [SpenceKonde/megaTinyCore#958] [Q&A] [answered]
Hi All, new to this so please pardon if I am not doing this correctly.

I have been playing around a little on-and-off with some attiny1616 boards for about a year now. Some of these are development boards and some I have designed and fabricated myself. Up to now I have not been having any major issues (once I finally figured out what in the world I was doing).

Today I went to test some new code on one of my development boards that I have been playing with successfully and got a slew of errors just trying to compile my code. I have done a whole lot of troubleshooting to see if I can figure this out on my own, including  removing and re-installing the entire board manager but nothing seems to work. I have gone all the way to the point where I have tried to compile a completely empty sketch and I am still getting the same errors. As far as I can tell I am doing everything the way I used to before (and yes, it has been a few months since I last did this so I may have forgotten something) so I am at a loss.

Since a picture is worth 1000 words here we go:

IDE Version:
![image](https://github.com/SpenceKonde/megaTinyCore/assets/28868163/07fb4056-d069-4dfe-8246-8d8a1af0258e)

Board Manager Version:
![image](https://github.com/SpenceKonde/megaTinyCore/assets/28868163/eff49ccd-4ab1-4f28-99b8-0b3d5eb88567)

Preferences:
![image](https://github.com/SpenceKonde/megaTinyCore/assets/28868163/2b2caa23-3615-4b49-9082-77fc0f257d5d)

My platform (in case it matters):
![image](https://github.com/SpenceKonde/megaTinyCore/assets/28868163/1abf9ef3-dfc8-459a-a65b-74b92eefd0fb)

Code:
```
#include <megaTinyCore.h>

void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}
```

I have tried both with and without the "#include" statement, it makes no difference.

My Board configuration:
![image](https://github.com/SpenceKonde/megaTinyCore/assets/28868163/20749227-42fd-4e59-b38a-e08b1781708f)

The errors I am getting:
```
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c: In function 'analogWrite':
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1205:13: error: 'set_inven' undeclared (first use in this function); did you mean 'strnlen'?
             set_inven = 1;
             ^~~~~~~~~
             strnlen
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1205:13: note: each undeclared identifier is reported only once for each function it appears in
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1307:5: error: break statement not within loop or switch
     break;
     ^~~~~
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1317:5: error: case label not within a switch statement
     case NOT_ON_TIMER: /*falls through*/
     ^~~~
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1318:5: error: 'default' label not within a switch statement
     default:
     ^~~~~~~
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1325:7: error: break statement not within loop or switch
       break;
       ^~~~~
In file included from /Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/api/ArduinoAPI.h:52:0,
                 from /Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/Arduino.h:26,
                 from /Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_private.h:32,
                 from /Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:25:
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c: At top level:
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/api/Common.h:16:26: error: expected ')' before numeric constant
 #define OUTPUT           1    /* used for pinMode() */
                          ^
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1330:16: note: in expansion of macro 'OUTPUT'
   pinMode(pin, OUTPUT);
                ^~~~~~
/Users/bdelapen/Library/Arduino15/packages/megaTinyCore/hardware/megaavr/2.6.7/cores/megatinycore/wiring_analog.c:1331:1: error: expected identifier or '(' before '}' token
 } // end of analogWrite
 ^
exit status 1
Error compiling for board ATtiny3226/3216/1626/1616/1606/826/816/806/426/416/406.
```

As you can tell, the code is not even compiling, I have tried with a board connected and without a board connected but as expected it makes no difference.

As mentioned above, I am at a loss and don't know what to do so any help would be greatly appreciated.

Birger

**Comment by bdelapen** [comment]
OK, quick update, I continued troubleshooting and I have now gotten things to compile and flash again by downgrading megaTinyCore to 2.6.5. Is this something that I should report as an issue instead?

**Comment by SpenceKonde** [comment, marked as answer]
Normally, yes, you should create an issue. In this one specific case, please don't, yours would be the 8th duplicate of the issue I closed, it is fixed in github - the last release broke 1-series parts and official Microchip boards fix will be in next release when it is tested and I can assure myself that it's tested because the CI will have been corrected as well. 

This went undetected because of deficiencies in the CI testing and a bug in the CLI regarding unspecified menus in an FQBN taking the value of "none of the above" instead of requiring it to take a value. That is often the default (generally not through intentional action, just the normal development process will tend to lead to submenus with nothing defined on their default setting. This wasn't such a case though. The upshot of it was that anything with a TCD (1-series) wouldn't compile if it was enabled for PWM and and nothing that didn't support the PWM mapping changes (ie, the microchip boards) would compile at all, and the CI missed both of those.

**Comment by bdelapen** [comment]
Thank you, I will close this discussion then.
